### PR TITLE
E-mail will be reflected on sign up and forgot password

### LINKED ISSUE #1264: Store the email address and reflect it everywhere in the app
**Actual Behaviour**

We have to type emails every time we switch to an activity after LoginActivity to ForgotPassword or back from it to LoginActivity.

**Expected Behaviour**

Store the email of the user in preferences and reflect it everywhere in the app, so that user does not need to type it everytime.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1264 

Changes: Now the email will be reflected on sign up and forgot password activity

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@batbrain7 Changes are not fulfilled

**Comment by batbrain7** [comment]
@iamareebjamal I only removed the merge conflict before,
pushed the fixes now please review

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 1]
```diff
@@ -41,12 +42,18 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
...
+                input_url_sign_up.visibility = View.VISIBLE
             } else {
-                input_url.visibility = View.GONE
+                input_url_sign_up.visibility = View.GONE
             }
         }
 
+        var bundle = intent.extras;
+        if (bundle != null) {
+            var string = bundle.getString("email")
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: This is a nullable field. There is no guarantee that `email` will be present in bundle

File: app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt [hunk 2]
```diff
@@ -41,6 +41,12 @@ class ForgotPasswordActivity : AppCompatActivity(), IForgotPasswordView {
             }
         }
 
+        var bundle = intent.extras;
+        if (bundle != null) {
+            var string = bundle.getString("email")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt]: There is no guarantee that email will be in this bundle, this should be nullable

File: app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt [hunk 3]
```diff
@@ -41,6 +41,12 @@ class ForgotPasswordActivity : AppCompatActivity(), IForgotPasswordView {
             }
         }
 
+        var bundle = intent.extras;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt]: Nullable

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 4]
```diff
@@ -40,13 +41,20 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
             email.editText?.setText(savedInstanceState.getCharSequenceArray(Constant.SAVED_STATES)[0].toString())
             password.editText?.setText(savedInstanceState.getCharSequenceArray(Constant.SAVED_STATES)[1].toString())
             confirm_password.editText?.setText(savedInstanceState.getCharSequenceArray(Constant.SAVED_STATES)[2].toString())
-            if (savedInstanceState.getBoolean(Constant.SERVER)) {
-                input_url.visibility = View.VISIBLE
+
+          if(savedInstanceState.getBoolean(Constant.SERVER)) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: Space after if

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 5]
```diff
@@ -40,13 +41,20 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
...
             } else {
-                input_url.visibility = View.GONE
+                input_url_sign_up.visibility = View.GONE
             }
         }
 
+        var bundle: Bundle? = intent.extras;
+        if (bundle != null) {
+            var string = bundle.getString("email")
+            email.editText?.setText(string)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: Same here

Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: @iamareebjamal where exactly the space? bcoz there is space after if here

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: > There is no guarantee that email will be in this bundle, this should be nullable

File: app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt [hunk 6]
```diff
@@ -41,6 +41,10 @@ class ForgotPasswordActivity : AppCompatActivity(), IForgotPasswordView {
...
+        var string = bundle?.getString("email")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt]: Still not nullable

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 7]
```diff
@@ -40,13 +42,18 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
...
-                input_url.visibility = View.VISIBLE
+                input_url_sign_up.visibility = View.VISIBLE
             } else {
-                input_url.visibility = View.GONE
+                input_url_sign_up.visibility = View.GONE
             }
         }
 
+        var bundle = intent.extras;
+        var string = bundle?.getString("email")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: Still not nullable

File: app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt (hunk 3 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt]: Should be val

File: app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt (hunk 6 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt]: Should be val

File: app/src/main/res/layout-land/activity_sign_up.xml [hunk 8]
```diff
@@ -69,14 +69,14 @@
         </android.support.design.widget.TextInputLayout>
 
         <android.support.v7.widget.AppCompatCheckBox
-            android:id="@+id/custom_server"
+            android:id="@+id/custom_server_sign_up"
```
Comment (iamareebjamal) [inline app/src/main/res/layout-land/activity_sign_up.xml]: Use camel case

File: app/src/main/res/layout-land/activity_sign_up.xml [hunk 9]
```diff
@@ -69,14 +69,14 @@
...
             android:layout_width="@dimen/activity_landscape_width"
             android:layout_height="wrap_content"
             android:layout_marginBottom="5dp"
             android:text="@string/custom_server" />
 
         <android.support.design.widget.TextInputLayout
-            android:id="@+id/input_url"
+            android:id="@+id/input_url_sign_up"
```
Comment (iamareebjamal) [inline app/src/main/res/layout-land/activity_sign_up.xml]: Use camel case

File: app/src/main/res/layout/activity_sign_up.xml [hunk 10]
```diff
@@ -69,14 +69,14 @@
         </android.support.design.widget.TextInputLayout>
 
         <android.support.v7.widget.AppCompatCheckBox
-            android:id="@+id/custom_server"
+            android:id="@+id/custom_server_sign_up"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/activity_sign_up.xml]: Use camel

File: app/src/main/res/layout/activity_sign_up.xml [hunk 11]
```diff
@@ -69,14 +69,14 @@
...
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_marginBottom="5dp"
             android:text="@string/custom_server" />
 
         <android.support.design.widget.TextInputLayout
-            android:id="@+id/input_url"
+            android:id="@+id/input_url_sign_up"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/activity_sign_up.xml]: Use camel case

File: app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt [hunk 12]
```diff
@@ -44,14 +43,14 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
...
-                input_url_sign_up.visibility = View.GONE
+                inputUrlSignUp.visibility = View.GONE
             }
         }
 
-        var bundle = intent.extras;
-        var string = bundle?.getString("email")
+        val bundle = intent.extras;
+        val string = bundle?.getString("email")
         email.editText?.setText(string)
```
[the remark this answers is not in the document]
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: @iamareebjamal if I check that string is not null then it ought to solve the problem

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt]: YES

File: app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt [hunk 13]
```diff
@@ -43,7 +43,10 @@ class ForgotPasswordActivity : AppCompatActivity(), IForgotPasswordView {
 
         val bundle = intent.extras;
         val string = bundle?.getString("email")
-        forgot_email.editText?.setText(string)
+        if (string != null)
+            forgot_email.editText?.setText(string)
+        else
+            forgot_email.editText?.setText("")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/forgotpassword/ForgotPasswordActivity.kt]: No need to set to empty, it'll automatically be empty

**Comment by batbrain7** [comment]
@iamareebjamal please merge this

**Pull request merged**